commands/run.go

```diff
@@ -461,3 +496,32 @@ var actorNamesToCodes = map[string]cid.Cid{
 	"fil/2/account":          builtin2.AccountActorCodeID,
 	"fil/2/multisig":         builtin2.MultisigActorCodeID,
 }
+
+type heightRange struct {
+	min, max int64
+}
+
+// divide divides a range into n equal subranges. The last range may be undersized.
+func (h heightRange) divide(n int) []heightRange {
+	size := h.max - h.min + 1 // +1 since the range is inclusive
+	if int64(n) > size {
+		panic(fmt.Sprintf("can't subdivide a range of %d into %d pieces", size, n))
+	}
+	step := size / int64(n)
+
+	subs := make([]heightRange, n)
+	subs[0].min = h.min
+	subs[0].max = h.min + step - 1
+
+	for i := 1; i < n; i++ {
+		subs[i].min = subs[i-1].max + 1
+		subs[i].max = subs[i].min + step
+	}
+
+	subs[n-1].max = h.max
+	return subs
+}
+
+func estimateCurrentEpoch() int64 {
+	return int64(time.Since(time.Date(2020, 8, 24, 22, 0, 0, 0, time.UTC)) / (30 * time.Second))
+}
```

Comment:
Not a fan of magic numbers.
```suggestion
var mainnetGenesis = time.Date(2020, 8, 24, 22, 0, 0, 0, time.UTC)
func estimateCurrentEpoch() int64 {
	return int64(time.Since(mainnetGenesis) / (builtin.EpochDurationSeconds))
}
```